SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+            shift
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+ENV_PATH="/home/$USER/${ENV_NAME}"
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if an environment with the same name was already created, if --clear is not passed
+if [ -d "$ENV_PATH" ] && [ -z "$CLEAR_ENV" ]; then
+    >&2 _log "ERROR: Virtual environment already exists."
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_CUSTOM_VERSION" ] && [ ! -e "$ACCPY_BASE/$ACCPY_CUSTOM_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version. Options: ${ACCPY_ALL_VERSIONS_STR}"
+    exit 1
+fi
+
+# Checks if a requirements file is given
+if [ -z "$REQUIREMENTS" ]; then
+    >&2 _log "ERROR: No requirements provided." && _log
+    print_help
+    exit 1
+# Checks if the provided requirements source is found
+elif [ -f $REQUIREMENTS ]; then
+    if [[ ${REQUIREMENTS##*.} != "txt" ]]; then
+        >&2 _log "ERROR: Invalid requirements file."
+        exit 1
+    fi
+    REQ_PATH=$REQUIREMENTS
+elif [[ $REQUIREMENTS == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REQUIREMENTS)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+    REPO_PATH=$HOME/SWAN_projects/${repo_name}
+
+    # Checks if the repository already exists in the home directory
+    if [ -d "${REPO_PATH}" ]; then
+        if [ -z "$CLEAR_ENV" ]; then
+            >&2 _log "ERROR: ${REPO_PATH} already exists. Use --clear to reclone the repository."
+            exit 1
+        else
+            rm -rf ${REPO_PATH}
+        fi
+    fi
+
+    echo "Cloning the repository from ${REQUIREMENTS}..."
+    # Clone the repository
+    git clone $REQUIREMENTS -q --template /usr/share/git-core/templates "${REPO_PATH}" || { >&2 _log "ERROR: Failed to clone repository"; exit 1; }
+
+    REQ_PATH=${REPO_PATH}/requirements.txt
+    # Check if requirements.txt exists in the repository
+    if [ ! -f "${REQ_PATH}" ]; then
+        rm -rf ${REPO_PATH}
+        >&2 _log "ERROR: requirements.txt not found in the repository"
+        exit 1
+    fi
+else
+    >&2 _log "ERROR: Requirements not found (${REQUIREMENTS})."
+    exit 1
+fi
+
+# Checks if the requirements file is not empty
+if [ ! -s "$REQ_PATH" ]; then
+    >&2 echo "ERROR: Requirements file is empty."
+    exit 1
+fi
+
+
+
+# --------------------------------------------------------------------------------------------
+
+
+# Credentials for the bash process to have access to EOS
+OAUTH2_FILE=$OAUTH2_FILE
+OAUTH2_TOKEN=$OAUTH2_TOKEN
+KRB5CCNAME=$KRB5CCNAME
+KRB5CCNAME_NB_TERM=$KRB5CCNAME_NB_TERM
+ACTIVATE_BIN_TEMPLATE="
```

Comment:
Do we still need this file? Wasn't this to make sure we correctly deactivated a a venv and restored the LCG environment? Since we don't do that anymore (one session, one env) I would remove this, and you can keep it somewhere in a branch of yours in case we need to rescue the code.

Reply:
I was talking about the ACTIVATE_BIN_TEMPLATE file you are creating here.